libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -3,19 +3,29 @@
 #include "utilities.hpp"
 
 #include <irods/MD5Strategy.hpp>
+#include <irods/dataObjInpOut.h>
 #include <irods/irods_hasher_factory.hpp>
 #include <irods/irods_log.hpp>
 #include <irods/irods_re_plugin.hpp>
 #include <irods/irods_re_ruleexistshelper.hpp>
+#include <irods/irods_rs_comm_query.hpp>
 #include <irods/rodsErrorTable.h>
 #include <irods/rsModAVUMetadata.hpp>
+#include <irods/library_features.h>
+#include <irods/irods_at_scope_exit.hpp>
 
 #define IRODS_QUERY_ENABLE_SERVER_SIDE_API
 #include <irods/irods_query.hpp>
 
-#define IRODS_IO_TRANSPORT_ENABLE_SERVER_SIDE_API
-#include <irods/dstream.hpp>
-#include <irods/transport/default_transport.hpp>
+#ifdef IRODS_HAS_FEATURE_ADMIN_MODE_FOR_DSTREAM_LIBRARIES
```

Comment:
This feature test macro name is just a placeholder until irods/irods#7530 is resolved.

This PR isn't blocked by that work. Once the irods/irods issue is resolved, these preprocessor macros can be updated to match the real feature test macro.